src/ssh/cockpitsshrelay.c

```diff
@@ -946,28 +947,40 @@ do_password_auth (CockpitSshData *data)
   return rc;
 }
 
+/* We don't support unlocking identities within cockpit-ssh so fail here */
 static int
-do_key_auth (CockpitSshData *data)
+prompt_for_identity_password (const char *prompt, char *buf, size_t len,
+                              int echo, int verify, void *userdata)
+{
+  CockpitSshData *data = (CockpitSshData *) userdata;
+  gchar *identity = NULL;
+  if (ssh_options_get (data->session, SSH_OPTIONS_IDENTITY, &identity) == SSH_OK)
+    {
+      data->problem_error = g_strdup_printf ("locked identity: %s", identity);
+      ssh_string_free_char (identity);
+    }
+  return -1;
```

Comment:
This looks like a memory leak now -- `identity` isn't used or passed to anywhere. If you don't need it for anything else than the error message, then move the `ssh_string_free_char` out of the `if`.